database/helpers.js

```diff
@@ -90,8 +90,14 @@ var addListing = function(id, userObj, callback) {
  */
 var filterByCity = function(city, callback) {
   var sql = `
-    SELECT * FROM profiles WHERE city = "${city}";`;
-  db.query(sql, (err, res) => (err ? callback(err) : callback(res)));
+    SELECT * FROM profiles INNER JOIN users ON users.id = profiles.userid  WHERE city = "${city}";`;
+  db.query(sql, function(err, res) {
+    if (err) {
+      throw err;
```

Comment:
Please handle errors appropriately (pass this error to the callback so that it is handled appropriately)